### PR TITLE
custommachine: update api: spec.master become plural, add required field 

### LINKED ISSUE #336: Add webhook for CustomCluster, CustomMachine, Fleet, Application, AttachedCluster
**What would you like to be added**:
add webhook for:
- [ ] CustomCluster
- [ ] CustomMachine
- [ ] Fleet
- [ ] Application
- [ ] AttachedCluster

**Why is this needed**:

At present, validation webhooks are only available for the CRD of the 'cluster.kurator.dev/v1alpha1' . 

However, validation webhooks are also required for the other CRDs.

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:

1 both `CustomMachineSpec.master ` and  `CustomMachineSpec.node` will become plural 
2 make these three variables `HostName` `PrivateIP` `PublicIP` required fields
3 add kubectl display for status.ready

It will be helpful to resolve #336 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
1 both `CustomMachineSpec.master ` and  `CustomMachineSpec.node` will become plural 
2 make these three variables `HostName` `PrivateIP` `PublicIP` required fields
3 add kubectl display for status.ready
```

### CODE REVIEW

File: pkg/apis/infra/v1alpha1/machine.go [hunk 1]
```diff
@@ -43,33 +42,29 @@ type CustomMachine struct {
 
 // CustomMachineSpec defines kubernetes cluster master and nodes.
 type CustomMachineSpec struct {
-	Master []Machine `json:"master,omitempty"`
-	Nodes  []Machine `json:"node,omitempty"`
+	Masters []Machine `json:"masters,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/infra/v1alpha1/machine.go:45]: This is a breaking change

Comment (Xieql) [reply pkg/apis/infra/v1alpha1/machine.go:45]: Or we could change them all to singular

Comment (Xieql) [reply pkg/apis/infra/v1alpha1/machine.go:45]: Current 'Master ' and 'Nodes  ' look strange because of inconsistent singular and plural forms."

**Pull request closed without merging**